sympy/integrals/tests/test_transforms.py

```diff
@@ -553,42 +751,26 @@ def test_laplace_transform():
         ((2*sin(s**2/(2*pi))*fresnelc(s/pi) - 2*cos(s**2/(2*pi))*fresnels(s/pi)
         + sqrt(2)*cos(s**2/(2*pi) + pi/4))/(2*s), 0, True))
 
+    # Matrix tests
     Mt = Matrix([[exp(t), t*exp(-t)], [t*exp(-t), exp(t)]])
     Ms = Matrix([[    1/(s - 1), (s + 1)**(-2)],
                  [(s + 1)**(-2),     1/(s - 1)]])
 
     # The default behaviour for Laplace tranform of a Matrix returns a Matrix
     # of Tuples and is deprecated:
     with warns_deprecated_sympy():
-        Ms_conds = Matrix([[(1/(s - 1), 0, Abs(s) > 1), ((s + 1)**(-2),
-            0, True)], [((s + 1)**(-2), 0, True), (1/(s - 1), 0, Abs(s) > 1)]])
+        Ms_conds = Matrix([[(1/(s - 1), 1, True), ((s + 1)**(-2),
+            -1, True)], [((s + 1)**(-2), -1, True), (1/(s - 1), 1, True)]])
     with warns_deprecated_sympy():
         assert LT(Mt, t, s) == Ms_conds
-
     # The new behavior is to return a tuple of a Matrix and the convergence
     # conditions for the matrix as a whole:
-    assert LT(Mt, t, s, legacy_matrix=False) == (Ms, 0, Abs(s) > 1)
-
+    assert LT(Mt, t, s, legacy_matrix=False) == (Ms, 1, True)
```

Comment:
I just want to note here that the `legacy_matrix=False` argument is a terrible API. I added it myself because there were backwards compatibility concerns with #21496 (https://github.com/sympy/sympy/pull/21496#discussion_r637461504). We should fix this so that a matrix is returned by default. Also the fact that `laplace_transform` sometimes returns an `Expr` and sometimes a `tuple` should be fixed somehow.

Reply:
The other integral transforms also do this, don't they?

Reply:
I'm not sure but it's a bad API. We shouldn't have tuples in a matrix and we also shouldn't have functions that sometimes return Expr and sometimes return tuples.